core/metadata/content.go

```diff
@@ -620,8 +620,20 @@ func (nw *namespacedWriter) Commit(ctx context.Context, size int64, expected dig
 	}); err != nil {
 		return err
 	}
+	if innerErr != nil {
+		return innerErr
+	}
 
-	return innerErr
+	if publisher := nw.db.Publisher(ctx); publisher != nil {
+		if err := publisher.Publish(ctx, "/content/create", &eventstypes.ContentCreate{
+			Digest: expected.String(),
```

Comment:
Either `dgst` should be declared to stay in scope here or expected updated after `nw.commit`. `expected` is not required to be set though.